### PR TITLE
gluon-config-mode-geo-location-osm: OpenStreetMap-based location picker

### PR DESCRIPTION
**Pull request by NeoRaider**
This PR contains the revised version of the OpenStreetMap-based location picker for the config mode (#1484).

Changes:
- Better integration with gluon-web
- OpenLayers 5 (instead of 2)
- Lots of changed strings (both package names and user-visible texts)
- Extend gluon-config-mode-geo-location rather than replacing it (avoiding to build multiple variants of the same code)

The one big remaining issue is that OpenLayers 5.1.3 does not work with XHTML (or HTML5 delivered with application/xhtml+xml Content-Type). This has been fixed in https://github.com/openlayers/openlayers/pull/8524, so I hope the next OpenLayers release will just work.

### CODE REVIEW

File: docs/user/site.rst [hunk 1]
```diff
@@ -407,6 +407,13 @@ config_mode \: optional
     package. Set *geo_location.show_altitude* to *true* if you want the altitude
     field to be visible.
 
+    The *geo_location.osm* section is only relevant when the *gluoin-config-mode-geo-location-osm*
```
Comment (2tata) [inline docs/user/site.rst]: gluoin-config-mode-geo-location-osm -> gluon-config-mode-geo-location-osm

File: package/gluon-config-mode-geo-location-osm/check_site.lua [hunk 2]
```diff
@@ -0,0 +1,4 @@
+need_number(in_site({'config_mode', 'geo_location', 'osm', 'center', 'lon'}))
+need_number(in_site({'config_mode', 'geo_location', 'osm', 'center', 'lat'}))
```
Comment (2tata) [inline package/gluon-config-mode-geo-location-osm/check_site.lua]: Maybe we can have default values like 0.0/0.0 to reduce site.conf parameters.

Comment (NeoRaider) [reply package/gluon-config-mode-geo-location-osm/check_site.lua]: Certainly not. This is a value that *must* be provided by each site, 0.0/0.0 simply doesn't make sense.

File: package/gluon-config-mode-theme/sass/gluon.scss [hunk 3]
```diff
@@ -284,6 +284,12 @@ code {
   margin-bottom: 2em;
 }
 
+.gluon-osm-map {
+  width: 100%;
+  height: 40em;
```
Comment (2tata) [inline package/gluon-config-mode-theme/sass/gluon.scss]: 40em seems to much maybe 25 is better the map looks nicer.

Comment (NeoRaider) [reply package/gluon-config-mode-theme/sass/gluon.scss]: The width of the main content scales between 40em and 60em (depending on the window width), meaning that with a height of 40em the map is square in narrow windows, and has a 3:2 aspect ratio in wide windows. This is already close to the usual widescreen display aspect ratio of 16:10.

Given that the relevant areas for sites should be circular on average, a square seems like the best approximation to me. Further decreasing the height would make the map less square, so we should either increase the height, or keep the current value...

**Comment by NeoRaider** [comment]
Screenshot:

image

**Comment by rotanid** [comment]
i think it's a nice solution.
btw., the fix to openlayers was merged.

**Comment by mweinelt** [comment]
This would be an ideal basis for a picking a domain, e.g. through boundary box overlays, like at https://darmstadt.freifunk.net/mitmachen/domains/.

**Comment by 2tata** [comment]
@NeoRaider There seems a bug in thins code. If I have "Advertise node position" checked and had already select a position on the map. After Pressing "Save & restart" the configmode will be just reload but not completed. Only after I have unchecked "Advertise node position" I can apply the formula.

**Comment by NeoRaider** [comment]
Hmm, I will look into this.

**Comment by NeoRaider** [comment]
I fixed two issues in the code:
* missing `validate()` override
* map was not using the previously submitted lon/lat values when form was rejected because of unrelated validation failures

**Comment by 2tata** [comment]
@NeoRaider I can confirm the bugfix. Now everything seems working as expected.

File: docs/user/site.rst [hunk 4]
```diff
@@ -407,6 +407,13 @@ config_mode \: optional
     package. Set *geo_location.show_altitude* to *true* if you want the altitude
     field to be visible.
 
+    The *geo_location.osm* section is only relevant when the *gluon-config-mode-geo-location-osm*
+    package is used. The *center.lon* and *center.lat* values are mandatory in this case and
+    define the default center of the map when no position has been picked yet. The *zoom* level
+    defaults to 12 in this case. *openlayers_url* allows to override the base URL of the
+    *build/ol.js* and *css/ol.css* files (the default is
+    ``https://cdn.rawgit.com/openlayers/openlayers.github.io/master/en/v5.1.3``).
```
Comment (2tata) [inline docs/user/site.rst]: url can be update to https://cdn.rawgit.com/openlayers/openlayers.github.io/master/en/v5.2.0/build/ol.js
that contains your upstream fixes.

**Comment by rubo77** [comment]
What is a valis URL? still the same as in the original PR?

        openlayers_url = 'http://dev.openlayers.org/OpenLayers.js',

**Comment by 2tata** [comment]
No its not. But you don't need to set a URL because a default value is already given. Furthermore there exists some documentation in docs/user/site.rst.

**Pull request merged**